src/apps/bm_devkit/serial_payload_example/user_code/user_code.cpp

```diff
@@ -81,8 +100,40 @@ void loop(void) {
     led1State = false;
   }
 
+  // Read a cluster of bytes if available
+  // -- A timer is used to try to keep clusters of bytes (say from lines) in the same output.
+  static int64_t readingBytesTimer = -1;
+  // Note - PLUART::setUseByteStreamBuffer must be set true in setup to enable bytes.
+  if (readingBytesTimer == -1 && PLUART::byteAvailable()) {
+    // Get the RTC if available
+    RTCTimeAndDate_t time_and_date = {};
+    rtcGet(&time_and_date);
+    char rtcTimeBuffer[32];
+    rtcPrint(rtcTimeBuffer, &time_and_date);
+    printf("[payload-bytes] | tick: %llu, rtc: %s, bytes:", uptimeGetMs(),
+           rtcTimeBuffer);
+    // not very readable, but it's a compact trick to overload our timer variable with a -1 flag
+    readingBytesTimer = (int64_t)((u_int32_t)uptimeGetMs());
+  }
+  while (PLUART::byteAvailable()) {
+    readingBytesTimer = (int64_t)((u_int32_t)uptimeGetMs());
+    uint8_t byte_read = PLUART::readByte();
+    printf("%02X ", byte_read);
+  }
+  if (readingBytesTimer > -1 &&
+      (u_int32_t)uptimeGetMs() - (u_int32_t)readingBytesTimer >= BYTES_CLUSTER_MS) {
+    printf("\n");
+    readingBytesTimer = -1;
+  }
+
   // Read a line if it is available
+  // Note - PLUART::setUseLineBuffer must be set true in setup to enable lines.
```

Comment:
what happens if you enable setUseLiineBuffer and setUserByteStreamBuffer? Do they work simultaneously?

Reply:
i see that they are both enabled by default, but I just want to confirm that they can work at the same time

Reply:
Yep - they don't stomp on each other. PLUART will process either or both depending on the configurations.